STDTC-99: Add onfilter function to ViewAllLogsFilter

UladzislauKutarkin opened the pull request:
[STDTC-99](https://issues.folio.org/browse/STDTC-99)

Here we add onFilter function to make search more consistent. Onfilter function give us possibility to search letter on any position of that letter, without it - it search only on first position 
On the attached video we can see how it works with onFilter and without on snapshot env

https://github.com/folio-org/stripes-data-transfer-components/assets/72550466/48d741ea-6511-4f1b-8e6b-5253836f180e

UladzislauKutarkin commented:
Here we just added `onFilter` function to `Selection` to make search more consistent

zburke reviewed (commented):
Sorry, I don't really understand what's happening here, and there's no additional detail in the Jira. I see your comment about making search "more consistent" but ... more consistent with what? 

Why do we need to add `onFilter` when this field already supports `onChange`? None of the other search elements here include both handlers so why does the job-profiles element need it?

zburke commented on lib/ViewAllLogsFilters/ViewAllLogsFilters.js:
```diff
@@ -85,6 +85,10 @@ export const ViewAllLogsFilters = ({
     label: `${user.firstName} ${user.lastName}`,
   })), 'value');
 
+  const onFilter = (value, dataOptions) => {
+    return dataOptions.filter(option => new RegExp(value, 'i').test(option.label));
+  };
+
```
Unit test? It should be dead-simple to write, and you can easily move this function out of the main component body since it's pure.

UladzislauKutarkin commented:
> Sorry, I don't really understand what's happening here, and there's no additional detail in the Jira. I see your comment about making search "more consistent" but ... more consistent with what?
> 
> Why do we need to add `onFilter` when this field already supports `onChange`? None of the other search elements here include both handlers so why does the job-profiles element need it?

I added video to explain changes and also changed the description

UladzislauKutarkin commented:
> Sorry, I don't really understand what's happening here, and there's no additional detail in the Jira. I see your comment about making search "more consistent" but ... more consistent with what?
> 
> Why do we need to add `onFilter` when this field already supports `onChange`? None of the other search elements here include both handlers so why does the job-profiles element need it?

We do such way in our modules, since it requirement from Magda. In onChange we make changes in the state of filters, `onFilter` props give us the possibility to make the filter work in the needed way
